### PR TITLE
feat(ethereum): < -adds new eth tx type

### PR DESCRIPTION
**Pull request by gskapka**
...per title.

The up and coming dencun fork implements EIP4844 which introduces a new receipt type for blob-carrying transactions.

This PR allows parsing of such transactions, and includes a new test with a goerli block containing the new type of txs.

### CODE REVIEW

**Comment by Oghma** [comment]
LGTM!

Some notes:
- This PR only updates the transaction type, block validation is still missing. Sentinel cannot validate a block an EIP4844 block.
- Reviewing the PR, I noticed that type 2 transactions are mislabelled. They should be EIP1559 and not EIP2718. The latter, introduces the field `TransactionType`.  EIP1559 introduces type 2 transactions with access list.

**Comment by gskapka** [comment]
Noted re point one. That will come in a following PR.

Re point two - fixing now, & will tag for re-review once ready.

**Review by Oghma — APPROVED** [review]
LGTM!

File: common/ethereum/src/eth_receipt_type.rs [hunk 1]
```diff
@@ -10,9 +10,9 @@ use strum_macros::EnumIter;
 #[derive(Clone, Debug, EnumIter, Eq, PartialEq, Serialize, Deserialize)]
 pub enum EthReceiptType {
     Legacy,
-    EIP2718, // NOTE: New tx type that's an envelope for future ones: https://eips.ethereum.org/EIPS/eip-2718
     EIP2930, // NOTE: Optional access lists: https://eips.ethereum.org/EIPS/eip-2930
     EIP4844, // NOTE: Dencun-fork blob-carrying txs: https://eips.ethereum.org/EIPS/eip-4844
+    EIP1559, // NOTE: New tx type that's an envelope for future ones: https://eips.ethereum.org/EIPS/eip-1559
```
[the remark this answers is not in the document]
Comment (gskapka) [reply common/ethereum/src/eth_receipt_type.rs]: fixed!

File: common/ethereum/src/eth_receipt_type.rs [hunk 2]
```diff
@@ -10,8 +10,9 @@ use strum_macros::EnumIter;
 #[derive(Clone, Debug, EnumIter, Eq, PartialEq, Serialize, Deserialize)]
 pub enum EthReceiptType {
     Legacy,
-    EIP2718,
-    EIP2930,
+    EIP2718, // NOTE: New tx type that's an envelope for future ones: https://eips.ethereum.org/EIPS/eip-2718
```
[the remark this answers is not in the document]
Comment (gskapka) [reply common/ethereum/src/eth_receipt_type.rs]: Fixed!

**Pull request merged**